server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1197,11 +1215,13 @@ public void parseXContent(XContentParser parser, boolean checkTrailingTokens) th
                     }
                 } else if (_SOURCE_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
                     fetchSourceContext = FetchSourceContext.fromXContent(parser);
+                    searchUsage.trackSectionUsage(_SOURCE_FIELD.getPreferredName());
```

Comment:
Do we need the extra logic here too?

Reply:
this is the `if token.isValue` branch. I don't think it should have additional logic. What is empty in that case? Invalid values will be rejected by the parser.

Reply:
Do we want to track it if we have `_source:true`, which is the default?

Reply:
I see what you mean, I addressed that . Hopefully there aren't any other cases like this one. This was one of the reasons why I did not add tracking for sort.